marge/approvals.py

```diff
@@ -17,7 +22,104 @@ def refetch_info(self):
         if gitlab_version.is_ee:
             self._info = self._api.call(GET(approver_url))
         else:
-            self._info = dict(self._info, approvals_left=0, approved_by=[])
+            self._info = self.get_approvers_ce()
+
+    def get_approvers_ce(self):
+        """get approvers status using thumbs on merge request
+        """
+        owner_file = self.get_codeowners_ce()
+        if not owner_file['owners']:
+            log.info("No CODEOWNERS file in master, continuing without approvers flow")
+            return dict(self._info, approvals_left=0, approved_by=[], codeowners=[])
+
+        code_owners = self.determine_responsible_owners(owner_file['owners'], self.get_changes_ce())
+
+        if not code_owners:
+            log.info("No matched code owners, continuing without approvers flow")
+            return dict(self._info, approvals_left=0, approved_by=[], codeowners=[])
+
+        awards = self.get_awards_ce()
+
+        up_votes = [e for e in awards if e['name'] == 'thumbsup' and e['user']['username'] in code_owners]
+
+        approvals_required = len(code_owners)
+
+        if owner_file['approvals_required'] > 0:
+            approvals_required = owner_file['approvals_required']
+
+        approvals_left = max(approvals_required - len(up_votes), 0)
+
+        return dict(self._info, approvals_left=approvals_left, approved_by=up_votes, codeowners=code_owners)
+
+    def determine_responsible_owners(self, owners_glob, changes):
+        owners = set([])
+
+        # Always add global users
+        if '*' in owners_glob:
+            owners.update(owners_glob['*'])
+
+        if 'changes' not in changes:
+            log.info("No changes in merge request!?")
+            return owners
+
+        for change in changes['changes']:
+            for glob, users in owners_glob.items():
+                test_glob = glob
+                if glob.endswith('/'):
+                    test_glob += "*"
+
+                if 'new_path' in change and fnmatch.fnmatch(change['new_path'], test_glob):
+                    owners.update(users)
+
+        return owners
+
+    def get_changes_ce(self):
+        changes_url = '/projects/{0.project_id}/merge_requests/{0.iid}/changes'
+        changes_url = changes_url.format(self)
+
+        return self._api.call(GET(changes_url))
+
+    def get_awards_ce(self):
+        emoji_url = '/projects/{0.project_id}/merge_requests/{0.iid}/award_emoji'
+        emoji_url = emoji_url.format(self)
+        return self._api.call(GET(emoji_url))
+
+    def get_codeowners_ce(self):
+        config_file = self._api.repo_file_get(self.project_id, "CODEOWNERS", "master")
```

Comment:
This should point to the project's `default_branch` from the API response instead a hardcoded master IMO (you have `develop`, `development`, `devel`, `main`, etc)

Reply:
I agree. I will have a look at this